addons/web_editor/static/src/js/editor/rte.summernote.js

```diff
@@ -1099,44 +1099,23 @@ var SummernoteManager = Class.extend(mixins.EventDispatcherMixin, ServicesMixin,
      */
     saveCroppedImages: function ($editable) {
         var defs = _.map($editable.find('.o_cropped_img_to_save'), async croppedImg => {
-            var $croppedImg = $(croppedImg);
-            $croppedImg.removeClass('o_cropped_img_to_save');
-
-            var resModel = $croppedImg.data('crop:resModel');
-            var resID = $croppedImg.data('crop:resID');
-            var cropID = $croppedImg.data('crop:id');
-            var mimetype = $croppedImg.data('crop:mimetype');
-            var originalSrc = $croppedImg.data('crop:originalSrc');
-
-            var datas = $croppedImg.attr('src').split(',')[1];
-            let attachmentID = cropID;
-            if (!cropID) {
-                var name = originalSrc + '.crop';
-                attachmentID = await this._rpc({
-                    model: 'ir.attachment',
-                    method: 'create',
-                    args: [{
-                        res_model: resModel,
-                        res_id: resID,
-                        name: name,
-                        datas: datas,
-                        mimetype: mimetype,
-                        url: originalSrc, // To save the original image that was cropped
-                    }],
-                });
-            } else {
-                await this._rpc({
-                    model: 'ir.attachment',
-                    method: 'write',
-                    args: [[cropID], {datas: datas}],
-                });
-            }
-            const access_token = await this._rpc({
-                model: 'ir.attachment',
-                method: 'generate_access_token',
-                args: [[attachmentID]],
+            croppedImg.classList.remove('o_cropped_img_to_save');
+
+            const croppedAttachmentSrc = await this._rpc({
+                route: '/web_editor/crop_attachment',
+                params: {
+                    res_model: croppedImg.dataset.resModel,
+                    res_id: parseInt(croppedImg.dataset.resId),
+                    name: croppedImg.dataset.originalName + '.crop',
+                    data: croppedImg.getAttribute('src').split(',')[1],
+                    mimetype: croppedImg.dataset.mimetype,
+                    original_id: parseInt(croppedImg.dataset.originalId),
+                },
+            });
```

Comment:
Yes. As you had mentioned, this is because the snippet containing the image could have been duplicated or saved and the only way to know if the image is used anywhere else is to parse all the pages :/

Reply:
Maybe add a comment in the code explaining that

Reply:
:+1: